rdr_service/resource/generators/participant.py

```diff
@@ -357,14 +367,24 @@ def _prep_modules(self, p_id, ro_session):
         consent_dt = None
 
         if results:
+            # Track the last module/consent data dictionaries generated, so we can detect and omit replayed responses
+            last_mod_processed = {}
+            last_consent_processed = {}
             for row in results:
+                consent_added = False
                 module_name = self._lookup_code_value(row.codeId, ro_session)
+                module_name = self._lookup_code_value(row.codeId, ro_session)
+                # Start with a default submittal status.  May be updated if this is a consent module with a specific
+                # consent question/answer that determines module submittal status
+                module_status = BQModuleStatusEnum.SUBMITTED
                 module_data = {
                     'module': module_name,
                     'baseline_module': 1 if module_name in self._baseline_modules else 0,  # Boolean field
                     'module_authored': row.authored,
                     'module_created': row.created,
                     'language': row.language,
+                    'status': module_status.name,
+                    'status_id': module_status.value
                 }
                 # Default status, may be updated based on consent answer
                 module_status = BQModuleStatusEnum.SUBMITTED
```

Comment:
This was deleted in the BQ code. Should it be deleted here?

Reply:
It should be added back in in the BQ code.  Lines 384-385?
It was due to the refactoring of some related logic, which needed the fields to be present in the dictionary by default.